src/qibo/backends/__init__.py

```diff
@@ -36,7 +36,8 @@ def __init__(self):
         if not self.check_availability(default_backend):  # pragma: no cover
             for backend in self.profile.get('backends'):
                 name = backend.get('name')
-                if self.check_availability(name):
+                if self.check_availability(name) and not backend.get('is_hardware'):
+                    # hardware backends cannot be default
```

Comment:
I am not sure I understand why this cannot be the default. By design it should be possible to have a lab server running exclusively qibolab. Maybe I am missing something?

Reply:
The only reason I added this here is that there are some tests in qibo for which the backend is not selected, so they fall back to the default backend, and use features that are not available in the hardware backend, such as density matrices, channels, etc.. So if someone uses an environment with hardware backends and no simulation backends other than numpy and the hardware backend is used as default, qibo tests will fail.

Summarizing this is only for development convenience. Once the hardware backend is developed enough to have all features it can support and proper testing mechanism and qibo tests are flexible enough to skip testing features that are not supported by hardware then I do not see any issue with setting hardware backend as default.

Reply:
Ok, thanks.